edk2toolext/versioninfo/versioninfo_helper.py

```diff
@@ -0,0 +1,556 @@
+# 	"FileFlags": "0",
+# 	"FileOS": "VOS_NT",
+# 	"FileType": "VFT_DRV",
+# 	"FileSubtype": "VFT2_DRV_SYSTEM",
+# 	"StringFileInfo": {
+# 		"CompanyName": "Example Company",
+# 		"ProductName": "Example Product",
+# 		"FileVersion": "1.0.0.0",
+# 	},
+# 	"VarFileInfo": {
+# 		"Translation": "0x0409 0x04b0"
+# 	}
+# }
+#
+# StringFileInfo and VarFileInfo can both have many more entries.
+# More information available at https://docs.microsoft.com/en-us/windows/win32/menurc/versioninfo-resource.
+class VERSIONINFOGenerator(object):
+    versionDict = None
+
+    def __init__(self, filepath):
+        try:
+            with open(filepath, "r") as jsonFile:
+                data = jsonFile.read()
+                try:
+                    self.versionDict = json.loads(data)
+                except json.decoder.JSONDecodeError as e:
+                    print(e, file=sys.stderr)
+
+        except FileNotFoundError:
+            print("ERROR: Could not find " + filepath, file=sys.stderr)
+
+    # Private, returns true if verStr is a valid version string, false otherwise.
+    def __validateVersionNumber__(self, verStr):
+        if verStr.count('.') != 3:
+            print("ERROR: Invalid version string: " + verStr + ". Version must be in form "
+                  + "\"INTEGER.INTEGER.INTEGER.INTEGER\".", file=sys.stderr)
+            return False
+
+        for substr in verStr.split("."):
+            try:
+                if int(substr) > 65535:
+                    print("WARNING: Integer overflow in version string: " + verStr + ".", file=sys.stderr)
+            except ValueError:
+                print("ERROR: Invalid version string: " + verStr + ". Version must be in form \" \
+                       INTEGER.INTEGER.INTEGER.INTEGER\".", file=sys.stderr)
+                return False
+
+        return True
+
+    # Checks if loaded JSON file represents a valid VERSIONINFO resource. Returns true if JSON
+    # is valid, print relevant error messages and returns false otherwise.
+    def validate(self):
+        if not self.versionDict:
+            return False
+
+        valid = True
+
+        # First Pass: Check to see if all required fields are present
+        required = {string.upper() for string in VERSIONFILE_REQUIRED_FIELDS}
+        allowed = {string.upper() for string in VERSIONFILE_ALLOWED_FIELDS}
+        for key in self.versionDict.keys():
+            if key.upper() not in allowed:
+                print("ERROR: Invalid parameter: " + key + ".", file=sys.stderr)
+                valid = False
+            else:
+                if key.upper() in required:
+                    required.remove(key.upper())
+
+        for remaining in required:
+            print("ERROR: Missing required parameter: " + remaining + ".", file=sys.stderr)
+            valid = False
+
+        if STRING_FILE_INFO_STR in self.versionDict:
+            requiredStringFileFields = copy.deepcopy(STRING_FILE_INFO_REQUIRED_FIELDS)
```

Comment:
another trick for this that doesn't require importing more code is to use the slice operation.  STRING_FILE_INFO_REQUIRED_FIELDS[:] will create a copy you can delete from.